lib/logstash/filters/memcached.rb

```diff
@@ -141,20 +164,21 @@ def do_set(event)
     return true
   end
 
-  def establish_connection
-    require 'dalli'
-
-    hosts = validate_connection_hosts
-    options = validate_connection_options
+  def new_connection(hosts, options)
     logger.debug('connecting to memcached', context(hosts: hosts, options: options))
-    Dalli::Client.new(@hosts, options).tap do |client|
-      begin
-        client.alive!
-      rescue Dalli::RingError
-        logger.error("failed to connect", context(hosts: hosts, options: options))
-        fail("cannot connect to memcached")
-      end
+    Dalli::Client.new(hosts, options).tap { |client| client.alive! }
+  end
+
+  def reconnect(hosts, options)
+    begin
+      @cache = new_connection(hosts, options)
+      @connected = true
+    rescue => e
+      logger.error("failed to reconnect to memcached", hosts: hosts, options: options, message: e.message)
+      event.tag(@tag_on_failure)
```

Comment:
we don't have `event` in scope here.

Reply:
good catch. definitely missing a spec for this too then.